src/escalus_session.erl

```diff
@@ -317,3 +336,15 @@ get_stream_management(Features) ->
 -spec get_advanced_message_processing(exml:element()) -> boolean().
 get_advanced_message_processing(Features) ->
     undefined =/= exml_query:subelement(Features, <<"amp">>).
+
+-spec get_sasl_mechanisms(exml:element()) -> features().
+get_sasl_mechanisms(Features) ->
+    MechCDataPath = [{element, <<"mechanisms">>}, {element, <<"mechanism">>}, cdata],
+    %% TODO: The expected modules are stored as a custom attribute,
+    %%       so they don't get optimized out by the compiler.
+    %%       By calling `?MODULE:module_info()` we force them to be read into
+    %%       the ERTS atom table from the attributes chunk.
+    ?MODULE:module_info(),
+    %% TODO: convert these to escalus_auth function names
+    [ {binary_to_existing_atom(Mech, utf8), true}
```

Comment:
TBH, I don't know where it's used. The plan was to leave it in a state which is good to go forward without changing the interface (i.e. format of `Features` returned), but I still have some time left so will look for places actually using this code.